kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js

```diff
@@ -681,6 +691,58 @@ define([
                 });
         }
 
+        /**
+         *
+         * @returns Promise that resolves into a whole bunch of validation objects
+         */
+        function doBulkValidation() {
+            // transform values to { paramId: [values] }
+            const paramValueMap = paramIds.reduce((valueMap, paramId) => {
```

Comment:
The basic order of operations here is:
1. Get all current model values as a map from paramId -> array of values
2. Get the validation option structure, with extra options for this special case (an error text change and making sure we look up object names)
3. Validate everything all at once.
4. Push those validations straight into the widgets to update their state.